addons/payment_authorize/models/payment_transaction.py

```diff
@@ -89,11 +89,18 @@ def _send_refund_request(self, refund_amount=None, create_refund_transaction=Tru
 
         Note: self.ensure_one()
 
-        :return: None
+        :param float refund_amount: The amount to refund
+        :param bool create_refund_transaction: Whether a refund transaction should be created or not
+        :return: The refund transaction if any, or `None`
+        :rtype: recordset of `payment.transaction`
         """
-        super()._send_refund_request(refund_amount=refund_amount, create_refund_transaction=False)
         if self.provider != 'authorize':
-            return
+            return super()._send_refund_request(
+                refund_amount=refund_amount, create_refund_transaction=create_refund_transaction
+            )
+        refund_tx = super()._send_refund_request(
+            refund_amount=refund_amount, create_refund_transaction=False
```

Comment:
Given that you change `refund_amount` to `amount_to_refund` in the action, I'd change it everywhere to avoid confusion and another bugfix PR because we keep having mismatches ;)